waterdata/location_utils.py

```diff
@@ -312,3 +313,74 @@ def parm_grp_sort(x):
     rollup_by_parameter_grp.pop('ALL', None)
 
     return {k: _extract_group_date_range(v) for k, v in rollup_by_parameter_grp.items()}
+
+
+def _summarize_site_collected_data(rolled_up_dataseries):
+    """
+    Take dataseries rolled up by parameter group and extract
+    the earliest year that data collection occurred and provide
+    a list of all distinct properties and species being measured.
+
+    :param dict rolled_up_dataseries: data series rolled-up by parameter group with certain groups we don't care about
+        excluded
+    :return: earliest year of data collection, list of distinct things being measured
+    :rtype: tuple
+
+    """
+    all_parms = list(itertools.chain.from_iterable([x['parameters'] for x in rolled_up_dataseries.values()]))
+    all_start_dates = [parm['start_date'] for parm in all_parms]
+    try:
+        start_year = min(all_start_dates).year
+    except ValueError:
+        start_year = None
+    # need to use a set -- there might be multiple variants for a measurements
+    # (e.g. "nitrite, water as N" vs "nitrite, water as NO3-")
+    short_parm_names = set([parm['parameter_name'].split(',')[0].upper() for parm in all_parms])
+    sort_top = ('DISCHARGE', 'GAGE HEIGHT', 'TEMPERATURE')
+    sorted_parm_names = sorted(short_parm_names, key=lambda x: (x not in sort_top, x))
```

Comment:
I see why this works. It will not work if we want TEMPERATURE before GAGE_HEIGHT for instance, but as long as it doesn't matter this works.

Reply:
I of thinking that I don't care about the order of those three items as long as they take priority over things like "Agency analyzing sample".